src/main/scala/io/iohk/ethereum/domain/Account.scala

```diff
@@ -30,4 +31,9 @@ case class Account(
 
   def withStorage(storageRoot: ByteString): Account =
     copy(storageRoot = storageRoot)
+
+  override def toString: String =
+    s"Account(nonce: $nonce, balance: $balance, " +
+      s"storageRoot: ${Hex.toHexString(storageRoot.toArray[Byte])}, codeHash: ${Hex.toHexString(codeHash.toArray[Byte])})"
```

Comment:
So I thought. The number of `Hex.toHexString(something.toArray[Byte])` is appalling. It would be nice to have our own type for `ByteString` and `Array[Byte]`, even if it's just a wrapper. 

But that's nothing that relates to this PR. Just complaining / throwing ideas.